Client/NotificationManager.swift

```diff
@@ -47,7 +47,27 @@ class NotificationManager {
         }
     }
 
+    @available(*, renamed: "requestAuthorization()")
+    func requestAuthorization(completion: @escaping (Result<Bool, Error>) -> Void) {
+        self.requestAuthorization { granted, error in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                completion(.success(granted))
+            }
+        }
+    }
+
+    func requestAuthorization() async throws -> Bool {
+        return try await withCheckedThrowingContinuation { continuation in
+            requestAuthorization { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
```

Comment:
This is marked as throwing, but we don't have a `continuation.resume(throwing....`

I could be not understanding the async/await pattern correctly, but, from what I know, we do want to explicitly throw from inside here, rather, than elsewhere.

Reply:
I can try to explain this concisely. In Swift asynchronous patterns, we have moved from using completion handlers with a format like (Value?, Error?) -> Void, to using (Result<Value>) -> Void, which simplified the code by leveraging the switch pattern. With the introduction of async/await, error handling is now performed with the do/catch pattern, leading to the use of `withCheckedThrowingContinuation`. In the case of `requestAuthorization`, an error may be returned in the Result error case. `withCheckedThrowingContinuation` provides an elegant solution by taking the result and returning an error in the catch block if it's in the error case.

Reply:
After reading [this article](https://www.donnywals.com/wrapping-existing-asynchronous-code-in-async-await-in-swift/), I realized I didn't know about `continuation.resume(with: result)` being a shorthand convenience function for
```
switch result {
            case .success(let token):
                continuation.resume(returning: token)
            case .failure(let error):
                continuation.resume(throwing: error)
            }
```
Which is where I was confused - because you weren't explicitly throwing an error, so a catch block wouldn't matter. But you are actually, by using the function that takes in a `Result` type

All good now. I learned something new about async/await. Awesome. ty

Reply:
I also didn't know you could do that 🤯